### ISSUE TITLE
CssSyntaxError on a valid line of code

### ISSUE THREAD

**Issue by SilencerWeb** [stylelint/stylelint#4634]
> Clearly describe the bug

I get a syntax error on a valid line of code

> What code is needed to reproduce the bug?

```jsx
export const OutermostOffset = styled.div`
  top: ${(p) => css`calc(${p.theme.viewportGapY} + constant(safe-area-inset-top))`};
`;
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "processors": ["stylelint-processor-styled-components"],
  "extends": [
    "stylelint-config-standard",
    "stylelint-config-styled-components"
  ],
  "rules": {
    "declaration-empty-line-before": null,
    "value-keyword-case": null
  }
}
```

> Which version of stylelint are you using?

13.2.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI, `stylelint ./src/**/style.ts`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it is related to styled-components. The problem is in `css` function. If I will remove it, then stylelint will work fine but this function is needed there

> What did you expect to happen?

No error

> What actually happened (e.g. what warnings or errors did you get)?

Got an error Unknown word, CssSyntaxError

**Comment by evilebottnawi** [comment]
Do not use `stylelint-processor-styled-components`, stylelint support styled-components out of box

/cc @chinesedfan please deprecate module, we get minimum 1-2 problems weekly

**Comment by SilencerWeb** [comment]
@evilebottnawi erm... even without using `stylelint-processor-styled-components` I get this error.

And tbh without `stylelint-processor-styled-components` I get even more errors, without this plugin stylelint doesn't support media queries

**Comment by evilebottnawi** [comment]
Yes, looks we have some problems on parser side https://github.com/stylelint/stylelint/issues/4574, let's reopen, but you do not use `stylelint-processor-styled-components`

**Comment by chinesedfan** [comment]
@evilebottnawi The processor is hard to be deprecated, until https://github.com/gucong3000/postcss-jsx/issues/62 gives some feedbacks or roadmaps. Thanks for understanding.

**Comment by evilebottnawi** [comment]
@chinesedfan we used fork version of postcss-jsx, please look https://github.com/stylelint/postcss-css-in-js, can you duplicate issue, thanks!

**Comment by jeddy3** [comment]
Closing as the [`syntax` option was removed](https://github.com/stylelint/stylelint/issues/5289) for the [14.0.0 release](https://github.com/stylelint/stylelint/issues/5205). (See the [migration guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#syntax-option-and-automatic-inferral-of-syntax).)

We'll also [be deprecating](https://github.com/stylelint/postcss-css-in-js/issues/225) our [forked CSS-in-JS package](https://github.com/stylelint/postcss-css-in-js), in favour of leaner custom syntaxes.

Please consider [writing a custom syntax](https://github.com/stylelint/stylelint/blob/main/docs/developer-guide/syntaxes.md) specific to the CSS-in-JS library in this issue and fixing the bug there.
